airflow/www/templates/airflow/ti_log.html

```diff
@@ -53,6 +53,7 @@ <h4>{{ title }}</h4>
     <div class="col-md-4 text-right">
       <a class="btn btn-default" onclick="scrollBottomLogs()">Jump To End</a>
       <a class="btn btn-default" onclick="toggleWrapLogs()">Toggle Wrap</a>
+      <a class="btn btn-default" onclick="downloadActiveLog()">Download</a>
```

Comment:
Can this be a “real” `<a>` tag `href` constructed when the tab is shown? Should probably add either `target` or [`download`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download) so the file doesn’t open in the current browser tab).

Reply:
@uranusjr Thanks for the suggestion. We can definitely do that.

It looks like the active "try number" is managed on the frontend, so one would still need to use javascript to render the correct url. An update might look like this:

#### ti_log.html
```html
<a class="btn btn-default" id="ti_log_download_active" download>Download</a>
```

#### ti_log.js
```js
function setDownloadUrl(tryNumber) {
  if (!tryNumber) {
    // default to the active tab value
    tryNumber = $('#ti_log_try_index_list .active a').attr('href').replace(/\D/g,'');
  }
  const query = new URLSearchParams({
    dag_id: dagId,
    task_id: taskId,
    execution_date: executionDate,
    try_number: tryNumber,
    metadata: 'null',
    format: 'file'
  });
  const url = `${logsWithMetadataUrl}?${query}`;
  $('#ti_log_download_active').attr('href', url);
}

$(document).ready(() => {
  // Set the download link once on initial page load
  setDownloadUrl();
  $('#ti_log_try_index_list a').click(function () {
    // Set it again after clicking a new try_number.
    // Obtain try_number from the clicked element because the
    // .active class has not been replaced yet.
    const tryNumber = $(this).attr('href').replace(/\D/g,'');
    setDownloadUrl(tryNumber);
  });
});
```

Regarding the `target` or `download` attributes, I chose to use neither for the following reasons:
- the log download links in the graph view task modal do not use them
- the flask handler sets the `Content-Disposition` header with the `attachment` parameter
- `download` doesn't seem to be [supported on IE](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#browser_compatibility)

Please let me know your thoughts, I'd be happy to make changes.

Reply:
If the server already sets `attachment` we can omit `download`, yes. Setting `href` as you outlined looks good to me. (Needs @bbovenzi to take another look.)